File: e-commerce/src/main/java/com/example/e_commerce/services/admin/AdminService.java
package com.example.e_commerce.services.admin;

import com.example.e_commerce.dto.CategoryDTO;
import com.example.e_commerce.dto.OrderDTO;
import com.example.e_commerce.dto.ProductDTO;
import com.example.e_commerce.entities.Category;
import com.example.e_commerce.entities.Product;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.List;


public interface AdminService {
    Category createCategory(CategoryDTO categoryDTO);

    List<CategoryDTO> getAllCategories();

    Product postProduct(Long categoryId, ProductDTO productDTO) throws IOException;

    List<ProductDTO> getAllProducts();

    void deleteProduct(Long Id);

    ProductDTO getProductById(Long Id);

    ProductDTO updateProduct(Long categoryId, Long productId, ProductDTO productDTO) throws IOException;

    List<OrderDTO> getAllOrders();

    List<OrderDTO> getAllOrders(Long userId);
}

File: ecommerce_web/src/app/admin/admin-service/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { LocalStorageService } from 'src/app/services/storage-service/local-storage.service';
import { environment } from 'src/environments/environment';

const BASIC_URL = environment['BASIC_URL'];
@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor( private http: HttpClient) { }

  postCategory(categoryDTO: any): Observable<any>{
    return this.http.post<[]>(BASIC_URL + "api/admin/category",categoryDTO,{
      headers: this.createAuthorizationHeader(),
    });
  }
  getAllCategories(): Observable<any>{
    return this.http.get<[]>(BASIC_URL + "api/admin/categories",{
      headers: this.createAuthorizationHeader(),
    });
  }
  postProduct(categoryId: number, productDTO: any): Observable<any>{
    return this.http.post<[]>(BASIC_URL + "api/admin/product" + categoryId ,productDTO,{
      headers: this.createAuthorizationHeader(),
    });
  }
  getAllProducts(): Observable<any>{
    return this.http.get<[]>(BASIC_URL + "api/admin/products",{
      headers: this.createAuthorizationHeader(),
    });
  }

  deleteProduct(id: number): Observable<any>{
    return this.http.delete<[]>(BASIC_URL + "api/admin/product/" + id,{
      headers: this.createAuthorizationHeader(),
    });
  }

  getProductById(id: number): Observable<any>{
    return this.http.get<[]>(BASIC_URL + "api/admin/product/" + id ,{
      headers: this.createAuthorizationHeader(),
    });
  }
  updateProduct(categoryId: number, productId: number,productDTO: any): Observable<any>{
    return this.http.put<[]>(BASIC_URL + `api/admin/${categoryId}/product/${productId}`,productDTO,{
      headers: this.createAuthorizationHeader(),
    });
  }
getAllOrders(): Observable<any>{
  return this.http.get<[]>(BASIC_URL + "api/admin/orders" ,{
    headers: this.createAuthorizationHeader(),
  });
}

  createAuthorizationHeader(): HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      `Authorization`,
      `Beare`+ LocalStorageService.getToken()
    )
  }
}
